### PR TITLE
Edge v5.0.0

### PR DESCRIPTION
**Pull request by emdobrin**
## Description

Updates minimum supported iOS, tvOS to 12.0
Fixes podspec verification failure
CircleCI, workflow Xcode updates
Readme updates

Note: If https://github.com/adobe/aepsdk-edgeidentity-ios/pull/117 is merged beforehand, I can update that dependency to dev-v5.0.0 also

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -83,14 +83,14 @@ commands:
     steps:
       - macos/preboot-simulator:
           platform: "iOS"
-          version: "16.1"
+          version: "17.0"
           device: "iPhone 14"
```
Comment (kevinlind) [inline .circleci/config.yml:87]: "iPhone 15" ?

**Review by kevinlind — COMMENTED** [review]
Looks good. One small change.

File: .circleci/config.yml (hunk 1 above)
Comment (cacheung) [inline .circleci/config.yml:87]: are we updating codecov to 3.3.0 for Edge?

File: Makefile [hunk 2]
```diff
@@ -150,7 +150,7 @@ lint-autocorrect:
 lint:
 	./Pods/SwiftLint/swiftlint lint Sources TestApps
```
Comment (cacheung) [inline Makefile:152]: There are instances for iPhone 14, need to change to iPhone 15

File: Script/test-podspec.sh [hunk 3]
```diff
@@ -55,18 +71,34 @@ rm -rf $PROJECT_NAME
...
-platform :tvos, '11.0'
+platform :tvos, '12.0'
 target '$PROJECT_NAME' do
   use_frameworks!
-  pod 'AEPCore', '~> 4'
-  pod 'AEPIdentity', '~> 4'
-  pod 'AEPLifecycle', '~> 4'
-  pod 'AEPServices', '~> 4'
-  pod 'AEPRulesEngine', '~> 4'
+  pod 'AEPCore', '~> 5'
```
Comment (cacheung) [inline Script/test-podspec.sh:97]: Should we use ~> 5.0? or 5.0.0?

File: Script/test-podspec.sh [hunk 4]
```diff
@@ -11,19 +11,35 @@ mkdir -p $PROJECT_NAME && cd $PROJECT_NAME
...
 target '$PROJECT_NAME' do
   use_frameworks!
-  pod 'AEPCore', '~> 4'
-  pod 'AEPIdentity', '~> 4'
-  pod 'AEPLifecycle', '~> 4'
-  pod 'AEPServices', '~> 4'
-  pod 'AEPSignal', '~> 4'
-  pod 'AEPRulesEngine', '~> 4'
+  pod 'AEPCore', '~> 5'
+  pod 'AEPIdentity', '~> 5'
```
Comment (cacheung) [inline Script/test-podspec.sh:38]: Should we use ~> 5.0 or 5.0.0

File: .circleci/config.yml (hunk 1 above)
Comment (emdobrin) [reply .circleci/config.yml]: I am seeing some issues with the codecov org and this macos/xcode version, and I am trying to confirm if these is a workaround with the Codecov team. we'll take any changes regarding codecov in a separate PR.

**Review by cacheung — APPROVED** [review]

**Review by kevinlind — APPROVED** [review]

**Pull request merged**